core/src/main/java/org/apache/calcite/rel/core/Project.java

```diff
@@ -442,6 +443,19 @@ public boolean isMapping() {
     return true;
   }
 
+  /**
+   * Given some columns, gets the non-constant ones.
+   */
+  public ImmutableBitSet getNonConstColumns(ImmutableBitSet columns) {
+    ImmutableBitSet.Builder nonConstCols = ImmutableBitSet.builder();
+    for (int col : columns) {
+      if (!RexUtil.isLiteral(getProjects().get(col), true)) {
+        nonConstCols.set(col);
+      }
+    }
```

Comment:
Would it better to put this method to RexUtils for being reused?

Reply:
Thanks for the good suggestion. I have moved it to class `RexUils`.